src/Dapr.PluggableComponents.Tests/DaprPluggableComponentsServiceBuilderTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Net.Sockets;
+using Dapr.PluggableComponents.Components;
+using Dapr.PluggableComponents.Components.StateStore;
+using Grpc.Net.Client;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Xunit;
+using static Dapr.Proto.Components.V1.StateStore;
+
+namespace Dapr.PluggableComponents;
+
+internal interface IMockStateStore : IStateStore
+{
+}
+
+internal sealed class MockStateStore : IMockStateStore
+{
+    private readonly IMockStateStore proxy;
+
+    public MockStateStore(IMockStateStore proxy)
+    {
+        this.proxy = proxy;
+    }
+
+    #region IStateStore Members
+
+    public Task DeleteAsync(StateStoreDeleteRequest request, CancellationToken cancellationToken = default)
+    {
+        return this.proxy.DeleteAsync(request, cancellationToken);
+    }
+
+    public Task<StateStoreGetResponse?> GetAsync(StateStoreGetRequest request, CancellationToken cancellationToken = default)
+    {
+        return this.proxy.GetAsync(request, cancellationToken);
+    }
+
+    public Task InitAsync(MetadataRequest request, CancellationToken cancellationToken = default)
+    {
+        return this.proxy.InitAsync(request, cancellationToken);
+    }
+
+    public Task SetAsync(StateStoreSetRequest request, CancellationToken cancellationToken = default)
+    {
+        return this.proxy.SetAsync(request, cancellationToken);
+    }
+
+    #endregion
+}
+
+public sealed class DaprPluggableComponentsServiceBuilderTests
+{
+    [Fact]
+    public async Task RegisterSingletonStateStore()
+    {
+        var mockStateStore = new Mock<IMockStateStore>();
+
+        var application = DaprPluggableComponentsApplication.Create();
+
+        application.Services.AddSingleton<IMockStateStore>(_ => mockStateStore.Object);
+
+        application.RegisterService(
+            "test", // TODO: Make unique.
+            serviceBuilder =>
+            {
+                serviceBuilder.RegisterStateStore<MockStateStore>();
+            });
+
+        await application.StartAsync();
+
+        string socketPath = "/tmp/dapr-components-sockets/test.sock";
```

Comment:
So this was something I was curious about: how to test this functionality without depending on the the filesystem.

I would suggest that instead of using a hardcoded folder, it would be better to use [a temporary directory](https://stackoverflow.com/questions/278439/creating-a-temporary-directory-in-windows). Not sure if .NET would expose `mkdtemp`.

Reply:
Fair. I was trying to keep things really simple to minimize the PR.  I've reafactored a bit to use a random socket file that's cleaned up post-test.